api.R

```diff
@@ -11,91 +12,130 @@ source("generics.R")
 # Read AWS creds from Docker secrets
 if (file.exists("/run/secrets/ENV_FILE")) {
   readRenviron("/run/secrets/ENV_FILE")
-} else {
+} else if (file.exists("secrets/ENV_FILE")) {
   readRenviron("secrets/ENV_FILE")
+} else {
+  readRenviron(".env")
 }
-readRenviron(".env")
 
 # Get the model run attributes at runtime from env vars
 dvc_bucket <- Sys.getenv("AWS_S3_DVC_BUCKET")
 run_bucket <- Sys.getenv("AWS_S3_MODEL_BUCKET")
-run_id <- Sys.getenv("AWS_S3_MODEL_RUN_ID")
-run_year <- Sys.getenv("AWS_S3_MODEL_YEAR")
 api_port <- as.numeric(Sys.getenv("API_PORT", unset = "3636"))
+default_run_id_var_name <- "AWS_S3_MODEL_RUN_ID"
+default_run_id <- Sys.getenv(default_run_id_var_name)
 
-
-# Download Files ---------------------------------------------------------------
-
-# Grab model fit and recipe objects
-temp_file_fit <- tempfile(fileext = ".zip")
-aws.s3::save_object(
-  object = file.path(
-    run_bucket, "workflow/fit",
-    paste0("year=", run_year),
-    paste0(run_id, ".zip")
-  ),
-  file = temp_file_fit
+# The list of run IDs that will be deployed as possible model endpoints
+valid_run_ids <- c(
+  "2024-02-06-relaxed-tristan",
+  "2024-03-17-stupefied-maya"
 )
 
-temp_file_recipe <- tempfile(fileext = ".rds")
-aws.s3::save_object(
-  object = file.path(
-    run_bucket, "workflow/recipe",
-    paste0("year=", run_year),
-    paste0(run_id, ".rds")
-  ),
-  file = temp_file_recipe
+assert_that(
+  default_run_id %in% valid_run_ids,
+  msg = sprintf(
+    "%s must be a valid run_id - got '%s', expected one of: %s",
+    default_run_id_var_name,
+    default_run_id,
+    paste(valid_run_ids, collapse = ", ")
+  )
 )
 
-# Grab metadata file for the specified run
-metadata <- read_parquet(
-  file.path(
-    run_bucket, "metadata",
-    paste0("year=", run_year),
-    paste0(run_id, ".parquet")
+# Given a run ID, return a model object that can be used to power a
+# vetiver API endpoint
+get_model_from_run_id <- function(run_id) {
+  run_year = substr(run_id, 1, 4)
+
+  # Download Files -------------------------------------------------------------
+
+  # Grab model fit and recipe objects
+  temp_file_fit <- tempfile(fileext = ".zip")
+  aws.s3::save_object(
+    object = file.path(
+      run_bucket, "workflow/fit",
+      paste0("year=", run_year),
+      paste0(run_id, ".zip")
+    ),
+    file = temp_file_fit
```

Comment:
The logic from here through the end of this function is identical to the previous version of the script, just factored out into the `get_model_from_run_id` function instead of applied to one single model.